ykbuild/build.rs

```diff
@@ -58,6 +59,25 @@ fn main() {
         }
     }
 
+    // We now know we want to build ykllvm. However, cargo can -- and in release mode does! -- run
+    // more than 1 copy of this build script in parallel. We thus need to make sure that we don't
+    // try and build multiple copies of ykllvm in the same directory at the same time, because that
+    // leads to the sort of hilarity where no-one laughs.
+    //
+    // To avoid parallel builds stomping on each other, we use a lock file to which we gain
+    // exclusive access: the first build script to do this wins, and any other build scripts that
+    // try to do so then sleep until the first has completed. File locking is a cross-platform
+    // nightmare, especially if you try to do something clever. Fortunately, we're not trying to
+    // make something clever like a read-write lock: we just gain exclusive (which probably means
+    // "write" on most platforms) access. Unix and Windows both remove the lock if the process dies
+    // unexpectedly (though Windows says, in essence, that it doesn't guarantee to do the unlocking
+    // very quickly), so if we fail while building, this should not cause parallel runs of this
+    // build script to deadlock.
+    let mut lock_path = ykllvm_dir.clone();
+    lock_path.push("build_lock");
+    let lock_file = File::create(lock_path).unwrap();
+    lock_file.lock_exclusive().unwrap();
```

Comment:
I think I may be able to kill this also. It's likely build-deps and run-deps building in parallel.

Reply:
Even if you can break those links (and I'm not sure it's possible), I would suggest keeping this: if anyone adds those links back in the future, this will stop themselves shooting them in the foot.